pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -566,10 +566,31 @@ func (r *ReconcileBareMetalHost) actionMatchProfile(prov provisioner.Provisioner
 		info.log.Info("updating hardware profile", "profile", hardwareProfile)
 		info.publishEvent("ProfileSet", fmt.Sprintf("Hardware profile set: %s", hardwareProfile))
 	}
+
 	info.host.ClearError()
 	return actionComplete{}
 }
 
+func updateRootDeviceHintsForHost(host *metal3v1alpha1.BareMetalHost) (dirty bool, err error) {
```

Comment:
Given that we expect more things to be stored in this way (RAID config, boot mode), would it be worth giving this a more generic name now to encourage that stuff to be added in this same function? Or do we want to keep this standalone and have another more generically-named function that calls it?

Reply:
I renamed it. I'm not 100% happy with the new name, so suggestions welcome.